typescript/index.ts

```diff
@@ -0,0 +1,48 @@
+export interface Workflow {
+  slug: WorkflowSlug;
+  name: string;
+  command: string;
+  tags?: [string];
+  description?: string;
+  arguments?: [Argument];
+  source_url?: string;
+  author?: string;
+  author_url?: string;
+  shells?: [Shell];
+}
+
+export interface Argument {
+  name: string;
+  description?: string;
+  default_value?: string;
+}
+
+export const enum Shell {
+  Bash,
+  Fish,
+  Zsh,
+}
+
+export type WorkflowSlug = string;
+
+function getFileNameOnly(filePath: string) : string {
+  return filePath.split("/").pop()!.split(".")!.shift()!;
+}
+
+const requireContext = require.context(
+  "yaml-loader!../specs",
```

Comment:
A note for the future, this assumes the directory of yaml files is flat and without any subfolders. I think this is fine for now

Reply:
yep, when we add subfolders we just need to change the `recursive` flag to `true` and it will traverse subdirectories as well.